cmd/kubectl-datadog/agent/check/check.go

```diff
@@ -0,0 +1,309 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package check
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-operator/pkg/plugin/common"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const maxParallel = 10
+
+var (
+	podName      string
+	node         string
+	checkExample = `
+  # check if the running Agents have detected check errors
+  %[1]s check
+
+  # check if the Agent foo has detected check errors
+  %[1]s check --pod-name foo
+
+  # check if the Agent running on node bar has detected check errors
+  # if both --pod-name and --node flags are present, the --node flag is ignored
+  %[1]s check --node bar
+`
+)
+
+// options provides information required by agent check command
+type options struct {
+	genericclioptions.IOStreams
+	configFlags   *genericclioptions.ConfigFlags
+	args          []string
+	clientset     *kubernetes.Clientset
+	restConfig    *restclient.Config
+	userNamespace string
+}
+
+// newOptions provides an instance of options with default values
+func newOptions(streams genericclioptions.IOStreams) *options {
+	return &options{
+		configFlags: genericclioptions.NewConfigFlags(false),
+		IOStreams:   streams,
+	}
+}
+
+// New provides a cobra command wrapping options for "check" sub command
+func New(streams genericclioptions.IOStreams) *cobra.Command {
+	o := newOptions(streams)
+	cmd := &cobra.Command{
+		Use:          "check [flags]",
+		Short:        "Find check errors",
+		Example:      fmt.Sprintf(checkExample, "kubectl datadog agent"),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.complete(c, args); err != nil {
+				return err
+			}
+			if err := o.validate(); err != nil {
+				return err
+			}
+			return o.run()
+		},
+	}
+
+	cmd.Flags().StringVarP(&podName, "pod-name", "p", "", "The Pod name of thee Agent to check")
+	cmd.Flags().StringVarP(&node, "node", "", "", "The node name where the Agent is running")
+
+	o.configFlags.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+// complete sets all information required for processing the command
+func (o *options) complete(cmd *cobra.Command, args []string) error {
+	o.args = args
+	var err error
+	o.restConfig, err = o.configFlags.ToRawKubeConfigLoader().ClientConfig()
+	if err != nil {
+		return fmt.Errorf("unable to instantiate restConfig: %v", err)
+	}
+	cmd.Println()
+	return common.SetupOptions(o, cmd)
+}
+
+// validate ensures that all required arguments and flag values are provided
+func (o *options) validate() error {
```

Comment:
I think it can be good to have:
```suggestion
func (o *options) validate(cmd *cobra.Command) error {
```
like this we can access all the useful methods in `Command`. I know it is not the case for now, but maybe in the futur.